pkg/controller/virtualmachineimport/virtualmachineimport_controller.go

```diff
@@ -118,146 +127,168 @@ func (r *ReconcileVirtualMachineImport) Reconcile(request reconcile.Request) (re
 	}
-	return nil, fmt.Errorf("Virtual machine %v not found", vmSpec.Name)
+	return nil, fmt.Errorf("Virtual machine %v not found", *vmSpec.Name)
 }
 
 func (r *ReconcileVirtualMachineImport) fetchOvirtSecret(vmImport *v2vv1alpha1.VirtualMachineImport) (*corev1.Secret, error) {
 	secret := &corev1.Secret{}
-	err := r.client.Get(context.TODO(), types.NamespacedName{Name: vmImport.Spec.Source.Ovirt.SecretName, Namespace: vmImport.Namespace}, secret)
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: vmImport.Spec.Source.Ovirt.ProviderCredentialsSecret.Name, Namespace: vmImport.Namespace}, secret) // FIXME:
 	return secret, err
 }
 
 func createOvirtConnection(secret *corev1.Secret) (*ovirtsdk.Connection, error) {
 	// TODO: CA cert
+	insecure, _ := strconv.ParseBool(string(secret.Data["insecure"]))
 	return ovirtsdk.NewConnectionBuilder().
 		URL(string(secret.Data["apiUrl"])).
 		Username(string(secret.Data["username"])).
 		Password(string(secret.Data["password"])).
-		Insecure(true).
+		Insecure(insecure).
 		Build()
 }
 
 // newDVForCR returns a VM specification of the fetched oVirt VM
-func newDVForCR(vm *ovirtsdk.Vm, secret *corev1.Secret) []cdiv1.DataVolume {
-	quantity, _ := resource.ParseQuantity("1Gi")
-	diskAttachments, _ := vm.DiskAttachments()
+func newDVForCR(diskAttachments *ovirtsdk.DiskAttachmentSlice, secret *corev1.Secret) []cdiv1.DataVolume {
+	quantity, _ := resource.ParseQuantity("1Gi") // FIXME:
 	dvs := make([]cdiv1.DataVolume, len(diskAttachments.Slice()))
-	for _, diskAttachment := range diskAttachments.Slice() {
+	for i, diskAttachment := range diskAttachments.Slice() {
 		disk, _ := diskAttachment.Disk()
-		dvs = append(dvs,
-			cdiv1.DataVolume{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      disk.MustId(), // FIXME:
-					Namespace: "default",     // FIXME:
-					Labels: map[string]string{ // FIXME:
-						"origin": "oVirt",
+		dvs[i] = cdiv1.DataVolume{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "cdi.kubevirt.io/v1alpha1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      diskAttachment.MustId(), // FIXME:
+				Namespace: "default",               // FIXME:
+				Labels: map[string]string{ // FIXME:
+					"origin": "oVirt",
+				},
+			},
+			Spec: cdiv1.DataVolumeSpec{
+				Source: cdiv1.DataVolumeSource{
+					Imageio: &cdiv1.DataVolumeSourceImageIO{
+						URL:           string(secret.Data["apiUrl"]),
+						DiskID:        disk.MustId(),
+						SecretRef:     "ovirt-key", // FIXME, should be created dynamically
+						CertConfigMap: "ovirt-ca",  // FIXME, should be created dynamically
 					},
 				},
-				Spec: cdiv1.DataVolumeSpec{
-					Source: cdiv1.DataVolumeSource{
-						/*
-							Imageio: &cdiv1.DataVolumeSourceImageIO{
-								URL:           secret.Data["apiUrl"],
-								DiskID:        vm.MustDiskAttachments().Slice()[0].MustDisk().MustId(),
-								SecretRef:     "",
-								CertConfigMap: "",
-							},
-						*/
+				PVC: &corev1.PersistentVolumeClaimSpec{
```

Comment:
TODO: Would be great to add `storageClassName` which we should get from the mapping.